GymBuddies/APIManagers/ParseAPIManager.m

```diff
@@ -44,4 +44,92 @@ +(void)logOut:(ParseManagerLogOutCompletionBlock) completion{
 }
 
 
++(void)fetchBodyZones:(ParseManagerFetchingDataRowsCompletionBlock) completion{
+    PFQuery *query = [PFQuery queryWithClassName:@"BodyZone"];
+    
+    ParseManagerFetchingDataRowsCompletionBlock block = ^void(NSArray *elements, NSError *error){
+        completion(elements, error);
+    };
+    
+    [query findObjectsInBackgroundWithBlock:block];
+}
+
+
++ (Exercise *)createExercise:(Exercise *)exercise
+           completion:(ParseManagerCreateCompletionBlock) completion {
+    
+    Exercise *newExercise = [Exercise initWithAttributes:exercise.title caption:exercise.caption author:exercise.author video:exercise.video image:exercise.image bodyZoneTag:exercise.bodyZoneTag];
+    
+    ParseManagerCreateCompletionBlock block = ^void(BOOL succeeded, NSError * _Nullable error){
+        completion(succeeded, error);
+        if(!succeeded){
+            return;
+        }
+    };
+    
+    [newExercise saveInBackgroundWithBlock:block];
+
+    [self saveExercise:newExercise completion:block];
+
+    return newExercise;
+}
+
+
++ (void)saveExercise:(Exercise *)exercise completion:(ParseManagerCreateCompletionBlock)completion{
+    PFObject *savedExercise = [PFObject objectWithClassName:@"SavedExercise"];
+    
+    savedExercise[@"author"] = exercise.author;
+    savedExercise[@"exercise"] = exercise;
+    
+    ParseManagerCreateCompletionBlock block = ^void(BOOL succeeded, NSError * _Nullable error){
+        completion(succeeded, error);
+        if(!succeeded){
+            return;
+        }
+    };
+    
+    [savedExercise saveInBackgroundWithBlock:block];
+}
+
+
++ (void)fetchUsersExercises:(ParseManagerFetchingDataRowsCompletionBlock) completion{
+    PFQuery *query = [PFQuery queryWithClassName:@"SavedExercise"];
+    [query includeKeys:@[@"exercise", @"exercise.author", @"exercise.bodyZoneTag", @"exercise.image"]];
+    [query whereKey:@"user" equalTo:[PFUser currentUser]];
+
+    ParseManagerFetchingDataRowsCompletionBlock block = ^void(NSArray *elements, NSError *error){
+        completion(elements, error);
+    };
+    
+    [query findObjectsInBackgroundWithBlock:block];
+    
+}
+
+
++ (PFFileObject *)getPFFileFromURL:(NSURL *)video{
+    if(!video){
+        return nil;
+    }
+    NSData *videoData = [NSData dataWithContentsOfURL:video];
+    // get image data and check if that is not nil
+    if (!videoData) {
+        return nil;
+    }
+    return [PFFileObject fileObjectWithName:@"video.mov" data:videoData];
```

Comment:
Is it a good idea to always use the same file name for video? Maybe you can check for file extension instead (so string match things that end with `.mov` and/or other video formats)